### ISSUE TITLE
Particular eslint warning not appeared

### ISSUE THREAD

**Issue by jackysee** [jose-elias-alvarez/null-ls.nvim#395]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

v0.5.1

### null-ls config

```lua
local null_ls = require("null-ls")
null_ls.config {
    debug = true,
    sources = {
        null_ls.builtins.diagnostics.eslint.with({
            only_local = "node_modules/.bin"
        })
    }
}
require("lspconfig")["null-ls"].setup { }
```

### Steps to reproduce

1. clone this repo https://github.com/jackysee/nvim-lsp-test
2. Run `npm install` to install a local eslint
3. Run `nvim -u null.vim .some.js`

### Expected behavior

A warning of `File ignored by default` should be shown. It's an implicit rule in eslint about linting dot file.
For comparison, you can run `nvim -u efm.vim .some.js`. The result should look like
![image](https://user-images.githubusercontent.com/30516/144528468-765af30c-3ce7-4adf-a7b6-4b44b1d9fd44.png)
 

### Actual behavior

No warning showed
![image](https://user-images.githubusercontent.com/30516/144528596-29134946-cb82-49ab-a7cb-b5557707918f.png)

### Debug log

[TRACE Fri 03 Dec 2021 09:17:57 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:121: received LSP request for method shutdown
[TRACE Fri 03 Dec 2021 09:17:57 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:146: received LSP notification for method exit
[TRACE Fri 03 Dec 2021 09:18:05 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:121: received LSP request for method initialize
[TRACE Fri 03 Dec 2021 09:18:05 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:146: received LSP notification for method initialized
[TRACE Fri 03 Dec 2021 09:18:05 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:146: received LSP notification for method textDocument/didOpen
[TRACE Fri 03 Dec 2021 09:18:05 AM HKT] ...plugged/vim-plug/null-ls.nvim/lua/null-ls/generators.lua:9: running generators for method NULL_LS_DIAGNOSTICS_ON_OPEN
[DEBUG Fri 03 Dec 2021 09:18:05 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/helpers.lua:447: attempting to find local executable node_modules/.bin/eslint
[DEBUG Fri 03 Dec 2021 09:18:05 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/helpers.lua:302: spawning command [/home/jackys/works/exp/nvim-lsp-test/node_modules/.bin/eslint] with args { "-f", "json", "--stdin", "--stdin-filename", "/home/jackys/works/exp/nvim-lsp-test/.some.js" }
[TRACE Fri 03 Dec 2021 09:18:06 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/helpers.lua:184: error output: nil
[TRACE Fri 03 Dec 2021 09:18:06 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/helpers.lua:185: output: [{"filePath":"/home/jackys/works/exp/nvim-lsp-test/.some.js","messages":[{"fatal":false,"severity":1,"message":"File ignored by default.  Use a negated ignore pattern (like \"--ignore-pattern '!<relative/path/to/filename>'\") to override."}],"errorCount":0,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":[]}]

[TRACE Fri 03 Dec 2021 09:18:06 AM HKT] ...lugged/vim-plug/null-ls.nvim/lua/null-ls/diagnostics.lua:140: received diagnostics from generators
[TRACE Fri 03 Dec 2021 09:18:06 AM HKT] ...lugged/vim-plug/null-ls.nvim/lua/null-ls/diagnostics.lua:141: {}
[TRACE Fri 03 Dec 2021 09:18:13 AM HKT] /tmp/plugged/vim-plug/null-ls.nvim/lua/null-ls/rpc.lua:146: received LSP notification for method textDocument/didClose

### Help

No

**Comment by jose-elias-alvarez** [comment]
Excellent reproduction, thank you! The issue was that the diagnostic did not specify a start line, which (mistakenly) caused the parser to discard the diagnostic as invalid. This should be fixed by 73420db2b58408a60a19024be3c14b0eba8413fe, so could you try updating and see if that takes care of it?

**Comment by jackysee** [comment]
Thank you~ It's fixed.
